src/commands/org/create/sandbox.ts

```diff
@@ -262,17 +293,42 @@ export default class CreateSandbox extends SandboxCommandBase<SandboxCommandResp
         this.flags.wait.seconds,
       ]);
     }
+
+    const sourceIdFlag = !!this.flags['source-id'];
+    const sourceSandboxNameFlag = !!this.flags['source-sandbox-name'];
+
+    const defFileContent = this.flags['definition-file'] ? readSandboxDefFile(this.flags['definition-file']) : {};
+    const sourceIdInDefFile = !!defFileContent.SourceId;
+    const sourceSandboxNameInDefFile = !!defFileContent.SourceSandboxName;
+
+    if ((sourceIdFlag || sourceSandboxNameFlag) && (sourceIdInDefFile || sourceSandboxNameInDefFile)) {
+      throw messages.createError('error.cloneFlagsConflict');
+    }
+  }
+
+  private async getSourceIdByName(): Promise<string | undefined> {
+    if (!this.flags['source-sandbox-name']) {
+      return undefined;
+    }
+    try {
+      const sourceOrg = await this.flags['target-org'].querySandboxProcessBySandboxName(
+        this.flags['source-sandbox-name']
+      );
+      return sourceOrg.SandboxInfoId;
+    } catch (err) {
+      throw messages.createError('error.noCloneSourceName', [this.flags['source-sandbox-name']], [], err as Error);
+    }
   }
 
   private async getSourceId(): Promise<string | undefined> {
-    if (!this.flags.clone) {
+    if (!this.flags['source-id']) {
       return undefined;
     }
     try {
-      const sourceOrg = await this.flags['target-org'].querySandboxProcessBySandboxName(this.flags.clone);
+      const sourceOrg = await this.flags['target-org'].querySandboxProcessBySandboxInfoId(this.flags['source-id']);
       return sourceOrg.SandboxInfoId;
     } catch (err) {
-      throw messages.createError('error.noCloneSource', [this.flags.clone], [], err as Error);
+      throw messages.createError('error.noCloneSourceId', [this.flags['source-id']], [], err as Error);
```

Comment:
this method can be removed, the value of `--source-id` or `sourceId` in the def. file is already a valid ID that we can pass to the API.